swagger.yaml

```diff
@@ -4137,6 +4138,48 @@ definitions:
         type: string
         description: The device path.
 
+  domains_resolve_request:
+    type: object
+    required:
+      - domains
+    properties:
+      domains:
+        type: array
+        items:
+          $ref: '#/definitions/domain-resolve-request'
+
+  domains_resolve_response:
+    type: object
+    required:
+      - domains
+    properties:
+      resolutions:
+        type: array
+        items:
+          $ref: '#/definitions/domain-resolve-response'
+
+  domain-resolve-request:
+    type: object
+    required:
+      - domain_name
+    properties:
+      domain_name:
+        description: "The domain name that should be resolved"
+        type: string
+
+  domain-resolve-response:
+    type: object
+    required:
+      - domain_name
+      - ip_address
```

Comment:
Will it always be a single IP?
Some DNSs handle multiple IPs and do the load balancing themselves.
I know this is not relevant for SNO, but in order for the API to not break, maybe it should be an array of IPs?
@avishayt what do you think?

Reply:
Good point. Just went over the `net.LookupIP` docs and they return a slice of IPs. The docs say "Returns a slice of the hostname's IPv4 and IPv6 **addresses**" but it's not clear whether it's just a single one of each or possibly multiple from each.

AFAIK the "load balancing" done by DNS server is hidden away from the DNS client - the server round robins (I think) the various IP addresses when returning a DNS response, the client doesn't see several. Maybe I'm wrong though.

In any case the fact that IPv6 can be returned means it should be an array as you said, with an additional enum field that says whether the result is an IPv4 or and IPv6.

Reply:
7f0feef

Fixed

Reply:
We should definitely handle this case.  It can be done with this API (have the same domain_name multiple times with different IPs).  This is simpler but less efficient.  What do you think?

Reply:
Nevermind, this new version looks good